### PR TITLE
solution to hw 10.1

### PR DESCRIPTION
**Pull request by syevtushok**

### CODE REVIEW

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 1]
```diff
@@ -17,7 +25,22 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        StringBuilder stringFromFile = new StringBuilder(Files.readString(Paths.get(fileName)));
+        Set<String> set = new TreeSet<>();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: let's put Characters in the set

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 2]
```diff
@@ -17,7 +24,23 @@
...
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        char[] chars = Files.readString(Paths.get(fileName)).toLowerCase()
+                .replaceAll("[^A-Za-z]", "").toCharArray();
+        Set<Character> set = new TreeSet<>();
+        for (char value : chars) {
+            set.add(value);
+        }
+        Iterator<Character> iterator = set.iterator();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: let's use `foreach` here

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 3]
```diff
@@ -17,7 +23,21 @@
...
+    public String getUniqueCharacters(String fileName) throws IOException {
+        char[] chars = Files.readString(Paths.get(fileName)).toLowerCase()
+                .replaceAll("[^A-Za-z]", "").toCharArray();
+        Set<Character> set = new TreeSet<>();
+        for (char value : chars) {
+            set.add(value);
+        }
+
+        StringBuilder letters = new StringBuilder();
+        for (char c : set) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]:
```suggestion
        for (char character : set) {
```

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 3 above)
Comment (syevtushok) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: done

**Pull request closed without merging**